blockprotocol/blockprotocol
# Fix Mock-Block-Dock

## 🌟 What is the purpose of this PR?

#826 updated the Graph Service for the new changes we're experimenting with in 0.3 These broke `MockBlockDock` (as expected). This PR updates `MockBlockDock` for the new definitions, giving it an understanding of the Type System, as well as implementing `Subgraph` resolution, creation, and exploration.

It implements the new format of the Graph Service hooks, and removes outdated code. 

There are a few driveby's for cleanliness too.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203363157432081/1203528958301159/f) _(internal)_

## 🚫 Blocked by




- [x] #826
- [x] #829

## 🔍 What does this change?

See commits, effort has been made to keep them self-explanatory

## 📜 Does this require a change to the docs?

- A full docs review will be needed when the 0.3 branch is more mature

## ⚠️ Known issues

I have tried to capture issues in code comments, or comments directly on the PR

## 🐾 Next steps

- Update MBD to support hooks for the ontology elements
- Update MBD to support temporal versioning

## 🛡 What tests cover this?

- Unknown, `tsc` is passing, and I have tested locally

## ❓ How to test this?

1.  Checkout the branch 
2.  Merge #829 in locally if it hasn't been merged upstream already (and this branch updated to include it)
3.  Run and test mock-block-dock as usual

**If you want to see the demo data** then change the following line in `useMockBlockProps`
```ts
entities: [...(initialEntities ?? initialMockData.entities)],
```
to this
```ts
entities: [...initialEntities ?? [], ...initialMockData.entities],
```

## 📹 Demo

Uploading Screen Recording 2022-12-11 at 03.50.14.mov…
